src/utils/openShiftConnectorUtils.ts

```diff
@@ -0,0 +1,77 @@
+import { commands, Disposable, Extension, extensions, ProgressLocation, window } from "vscode";
+
+export const OPENSHIFT_CONNECTOR_EXTENSION_ID: string = 'redhat.vscode-openshift-connector';
+export const OPENSHIFT_CONNECTOR = 'OpenShift Connector extension';
+const DOWNLOAD_TIMEOUT: number = 60000; // Timeout for downloading VSCode OpenShift Connector, in milliseconds
+
+/**
+ * Returns true if the OpenShift connector extension is installed, and false otherwise
+ *
+ * @returns true if the OpenShift connector extension is installed, and false otherwise
+ */
+export function isOpenShiftConnectorInstalled(): boolean {
+  return !!extensions.getExtension(OPENSHIFT_CONNECTOR_EXTENSION_ID);
+}
+
+/**
+ * Returns the OpenShift Connector extension API
+ *
+ * @throws Error if the extension is not installed
+ * @returns the OpenShift Connector extension API
+ */
+export async function getOpenShiftConnector(): Promise<any> {
+  if (!isOpenShiftConnectorInstalled()) {
+    throw new Error(`${OPENSHIFT_CONNECTOR} is not installed`);
+  }
+  const openShiftConnector: Extension<any> = extensions.getExtension(OPENSHIFT_CONNECTOR_EXTENSION_ID);
+  if (openShiftConnector.isActive) {
+    return openShiftConnector.exports;
+  }
+  return extensions.getExtension(OPENSHIFT_CONNECTOR_EXTENSION_ID).activate();
+}
+
+/**
+ * Install the OpenShift Connector extension
+ *
+ * @returns when the extension is installed
+ * @throws if the user refuses to install the extension, or if the extension does not get installed within a timeout period
+ */
+async function installOpenShiftConnector(): Promise<any> {
+  const installDisposables: Disposable[] = [];
+  try {
+    return Promise.race([
+      new Promise((resolve, reject) => {
+        extensions.onDidChange(() => {
+          if (isOpenShiftConnectorInstalled()) {
+            installDisposables.forEach((d: Disposable) => { d.dispose(); });
+            resolve();
+          }
+        }, null, installDisposables);
+        commands.executeCommand("workbench.extensions.installExtension", OPENSHIFT_CONNECTOR_EXTENSION_ID)
+          .then((_unused: any) => { }, reject);
+      }),
+      new Promise((_unused, reject) => {
+        // Fail install if it takes longer than the timeout period
+        setTimeout(reject, DOWNLOAD_TIMEOUT, new Error(`${OPENSHIFT_CONNECTOR} installation is taking a while. Cancelling 'Deploy to OpenShift'`));
+      })
+    ]);
+  } catch (e) {
+    installDisposables.forEach((d: Disposable) => { d.dispose(); });
```

Comment:
The code creates Promise instance and returns it. This catch block is basically dead branch IMO, it will never get control. Exceptions inside callbacks will be caught at line 50 in `deployToOpenShift.ts`.

Reply:
Good catch, thanks! I'll fix this later today